src/tree/SubscriptionTreeItem.ts

```diff
@@ -76,6 +81,26 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
         return result.data;
     }
 
+    public async pickResourceGroup(context: IActionContext, options: PickResourceGroupOptions & { canPickMany: true }): Promise<ResourceGroupTreeItem[]>;
+    public async pickResourceGroup(context: IActionContext, options: PickResourceGroupOptions & { canPickMany: false }): Promise<ResourceGroupTreeItem>;
+    public async pickResourceGroup(context: IActionContext, options: PickResourceGroupOptions): Promise<ResourceGroupTreeItem | ResourceGroupTreeItem[]> {
+        await this.getCachedChildren(context);
+        const rgsTreeMap = this.cache.getTreeMap(context, 'resourceGroup');
```

Comment:
I think rather than making the tree map, we can leverage the ResourceGroups on the ResourceCache.  

We can just make sure that it is resolved whenever this command is called.  I don't think you would even need the tree item for deleting the ResourceGroup.

This method certainly works, but it makes the getTreeMap function a little hacky and feels like a bit of a workaround.

Reply:
I think it would make sense for the SubscriptionTreeItem to have a `getResourceGroupsTask` promise that we can just check and re-resolve on refreshes

Reply:
> I think rather than making the tree map, we can leverage the ResourceGroups on the ResourceCache.
>
> We can just make sure that it is resolved whenever this command is called. I don't think you would even need the tree item for deleting the ResourceGroup.

Although we can delete the RG without the `ResourceGroupTreeItem`, in order to show a temporary description and a spinner on the resource group during delete, we need to call `.deleteTreeItem`.

Reply:
That would only be in the case if the groupBy setting was already set to ResourceGroup, right?  Otherwise it wouldn't be displayed in UI anyway.

In that case, you could take the RG and if you find the `ResourceGroupTreeItem`, do `.deleteTreeItem`; otherwise just delete it via the API call.

I don't have a very strong opinion about this, but it doesn't really feel like how `getTreeMap` should be used especially since you're calling `getCachedChildren` which would also call `getTreeMap`.  I'm worried that this code might mess with the `AppResourceTreeItems`

```
public mapSubGroupConfigTree(context: IActionContext,
        groupBySetting: string,
        treeMap: GroupTreeMap,
        getResourceGroup: (resourceGroup: string) => Promise<ResourceGroup | undefined>): void {

        const configId: string | undefined = this.groupConfig[groupBySetting]?.id.toLowerCase() ?? `${this.rootGroupConfig.id}/${ungroupedId}`;
        let subGroupTreeItem = treeMap[configId];
        if (!subGroupTreeItem) {
            subGroupTreeItem = this.createSubGroupTreeItem(context, groupBySetting, getResourceGroup);
            treeMap[configId] = subGroupTreeItem;
        }

        subGroupTreeItem.treeMap[this.id] = this;
        void ext.appResourceTree.refreshUIOnly(subGroupTreeItem);
    }
```

Reply:
I don't think it does since it doesn't actually set anything on the `AppResourceTreeItem` and treeMap is something is passed in, but that could definitely change in the future.

Reply:
I don't love the fact that different things would be happening based on the group by setting. And also if they switched to grouping by resource group after initiating a delete, the deleting... state wouldn't be reflected. That's definitely a nit at this point, but I can see how that isn't the best UX.

I could make a new method that is responsible for creating the RG tree items and caching them... But I want to avoid a large change like that so close to release.

Reply:
What if we change 
```
public set resourceGroups(resourceGroups: ResourceGroup[]) {
        const { addedResources, deletedResources } = this.diffProvidedResources(this._resourceGroups, resourceGroups);
        this.deleteResourcesFromCache(this._resourceGroups, deletedResources);
        this._resourceGroups.push(...addedResources);
    }
```

to store the `ResourceGroupTreeItems` rather than the `ResourceGroup` exactly like `appResources` does?